### PR TITLE
[217273] REST API updates

### PR DESCRIPTION
**Pull request by pkujawa**
[AB#217273](https://unicef.visualstudio.com/4e044e8d-bc28-4768-8074-f80ff6e4a0e5/_workitems/edit/217273)
[AB#217555](https://unicef.visualstudio.com/4e044e8d-bc28-4768-8074-f80ff6e4a0e5/_workitems/edit/217555)

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -25,3 +25,15 @@ coverage:
     - "hct_mis_api/conftest.py"
     - "hct_mis_api/config/settings.py"
     - "hct_mis_api/apps/core/management/commands/*"
+    - "pragma: no cover"
+    - "pragma: no-cover"
+    - "def __repr__"
+    - "if self\.debug"
+    - "raise AssertionError"
+    - "raise NotImplementedError"
```
[the remark this answers is not in the document]
Comment (pkujawa) [reply codecov.yml:33]: It is supposed to match if `self.debug:`
It was taken from the original file that was not moved but I actually think we do not have it anywhere in the code anymore. So will remove

File: src/hct_mis_api/api/endpoints/lookups/base.py [hunk 2]
```diff
@@ -25,9 +26,12 @@ def get(self, request: "Request", format: Optional[Any] = None) -> Response:
         return Response(dict(IDENTIFICATION_TYPE_CHOICE))
 
 
-class Country(HOPEAPIView):
+class CountryAPIView(HOPEAPIView):
+    serializer_class = CountrySerializer
+
     def get(self, request: "Request", format: Optional[Any] = None) -> Response:
-        return Response(dict(Countries()))
+        serializer = self.serializer_class(Country.objects.all(), many=True)
```
Comment (domdinicola) [inline src/hct_mis_api/api/endpoints/lookups/base.py:33]: let's change this in a viewset @pkujawa

**Review by MarekBiczysko — APPROVED** [review]

**Review by pavlo-mk — APPROVED** [review]
LGTM

File: src/hct_mis_api/api/endpoints/lookups/base.py (hunk 2 above)
Comment (pkujawa) [reply src/hct_mis_api/api/endpoints/lookups/base.py:33]: @domdinicola but we only want to retrieve list, right? All the lookups in this file are constructed this way

File: src/hct_mis_api/api/endpoints/program/views.py [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
+    from rest_framework.request import Request
+
 
 class ProgramGlobalListView(HOPEAPIView, ListAPIView):
     serializer_class = ProgramGlobalSerializer
     queryset = Program.objects.all()
+
+    def list(self, request: "Request", *args: Any, **kwargs: Any) -> Response:
+        queryset = self.queryset
+        filterset = ProgramFilter(request.GET, queryset=queryset)
```
Comment (domdinicola) [inline src/hct_mis_api/api/endpoints/program/views.py:21]: @pkujawa can we user filterset_class and pagination_class @pkujawa

**Pull request merged**